Lib/unittest/test/testmock/testasync.py

```diff
@@ -474,27 +521,31 @@ def __aiter__(self):
 
             raise StopAsyncIteration
 
-    def test_mock_aiter_and_anext(self):
-        instance = self.WithAsyncIterator()
-        mock_instance = MagicMock(instance)
-
-        self.assertEqual(asyncio.iscoroutine(instance.__aiter__),
-                         asyncio.iscoroutine(mock_instance.__aiter__))
-        self.assertEqual(asyncio.iscoroutine(instance.__anext__),
-                         asyncio.iscoroutine(mock_instance.__anext__))
-
-        iterator = instance.__aiter__()
-        if asyncio.iscoroutine(iterator):
-            iterator = asyncio.run(iterator)
-
-        mock_iterator = mock_instance.__aiter__()
-        if asyncio.iscoroutine(mock_iterator):
-            mock_iterator = asyncio.run(mock_iterator)
+    def test_aiter_set_return_value(self):
+        mock_iter = AsyncMock(name="tester")
+        mock_iter.__aiter__.return_value = [1, 2, 3]
+        async def main():
+            async for i in mock_iter:
```

Comment:
Slightly confused over if this is needed since we just return a list comprehension and assert only once.

Reply:
Ah, you're right, didn't notice the double loop